### ISSUE TITLE
Basic support to Postgresql

### ISSUE THREAD

**Issue by pedro-stanaka** [dereuromark/cakephp-queue#106]
Hello, 

I've tried to use the plugin with postgres and it seems to have an error on the ```QueuedTasksTable```. 
The query expression written is only compatible with MySQL in the following part:

```php
<?php
      public function requestJob(array $capabilities, $group = null) {
		$now = new Time();
		$nowStr = $now->toDateTimeString();

		$query = $this->find();
		$findCond = [
			'conditions' => [
				'completed IS' => null,
				'OR' => [],
			],
			'fields' => [
				'age' => $query->newExpr()->add('IFNULL(TIMESTAMPDIFF(SECOND, "' . $nowStr . '", notbefore), 0)')
			],
			'order' => [
				'age ASC',
				'id ASC',
			]
		];
```

If you instruct me on how I could solve this I can send a PR. This is somehow a pretty difficult expression to be written in a generic way for all databases. 
In postgres it would be something like this:

```sql
SELECT IFNULL(EXTRACT(EPOCH FROM ( notbefore - NOW() ), 0)
FROM queued_tasks
```

**Comment by dereuromark** [comment]
Would be nice to have travis CI for the queue. So far it is only tests on MySQL.

Regarding the query, we can also implement a switch based on the type if we can't figure out a good solution.
Usually @jose_zap is the master of ORM :P Especially for this kind of query.

**Comment by dereuromark** [comment]
I added it to the test matrix for now as allow_failure: https://travis-ci.org/dereuromark/cakephp-queue/jobs/169676171
There might be other small issues to be fixed.

**Comment by pedro-stanaka** [comment]
@dereuromark After a more thorough study of the class I saw a big number of queries that will have to be rewritten to work on PostgreSQL. I am working on it on my fork, but right now I don't know if it is a good idea to work with switches based on Driver type. 

I was thinking of just issuing a very basic query and then do some of the hard work on the application using features like MapReduce of the ORM, but any better ideas are appreciated.

**Comment by dereuromark** [comment]
Any update here? Maybe someone has the time and can do the postgres support?

**Comment by pedro-stanaka** [comment]
I actually began to work on it in my fork, but as soon as I realized the amount of queries that would need to be rewritten I put it on hold. It is too much effort to make it work only with postgres, if someone needs to work on other database then the person needs to rewrite everything again. My solution was to create a way simpler method using crontab. Since I am the opener of this issue, if you want @dereuromark you can close it.

**Comment by dereuromark** [comment]
Well, we can also part by part make the tests pass, and make it work with postgres.
We dont have to make it all work at once.
So feel free to PR.

**Comment by dereuromark** [comment]
Should only be 2 or 3 queries that really need some adjustments:
https://travis-ci.org/dereuromark/cakephp-queue/jobs/176869139

**Comment by charlieboygis** [comment]
Hi, very smart lib.
I made an attempt to have PG support, you can look at my forked version [here](https://github.com/charlieboygis/cakephp-queue/blob/master/src/Model/Table/QueuedJobsTable.php)

**Comment by dereuromark** [comment]
Nice! You can make a PR once it is ready to be merged. Let me know if you need any help.

**Comment by dereuromark** [comment]
@charlieboygis Any update here? Can you please also check the latest PR regarding the duplicate issue for PG support?

**Comment by dereuromark** [comment]
Currently tests are not failing.
Does anyone want to take another stab here to fully integrate PostgreSQL?
